### DISCUSSION TITLE
Proposal to migrate swmmio visualization features to separate package

### DISCUSSION THREAD

**Discussion by aerispaha** [pyswmm/swmmio#214] [Ideas]
Hi friends! 

Since `swmmio` was started almost 8 years ago, it's scope has evolved in interesting ways. It has been so nice see how our users have leveraged it in their own projects and research, and to see it's functionality grow from community contributions ❤️.  

One thing that I notice, however, is that our visualization feature set has gotten fairly dated. And, based on user contributions and feedback, my estimation is that these features aren't used that often. Assuming I'm not way off, I'd like to propose that we refocus `swmmio` on what I think it does best (reading, writing, and editing SWMM inp and rpt files) while gradually deprecating the visualization features. 

What do y'all think about that? Are there many users that are leveraging the `swmm_graphics` module?

**Comment by bemcdonnell** [comment]
@aerispaha This is a really thoughtful proposal.  There is a new parser coming out (hopefully soon) that will bring much cleaner coverage to the INP file.  That will really elevate this project!  I think moving the Viz features out will be really good from the perspective of the integration ecosystem that will be announced very soon.   What do you think are the next steps?

**Comment by bowguy** [comment]
I have just started an extensive use of pyswmm for flood forecasting.  In addition to getting the data from pyswmm, I am leveraging the create_map function from swmmio.  I would love to continued as it is exactly what I need.  I have have a few comments/suggestions but it is incredibly valuable to me.
create_map is now throwing lots of warning messages in Python3.11 BTW

**Reply by bemcdonnell** [reply]
Do tell!

**Reply by bowguy** [reply]
LOL.  I would love to show you but right now, there is no rain predicted for the next few days so the output is pretty barren
Check out:
https://mhfd-ns5data.trilynx-novastar.systems/mhfd/data/json/MWPS/Lena_107861_fcst_test.html
for an archived run

**Reply by bemcdonnell** [reply]
@bowguy Can you email me directly? I'm actually super interested in what MHFD is up to (and I'm a near neighbor up in Boulder :) )
